src/main/java/net/snowflake/client/core/ArrowSqlInput.java

```diff
@@ -205,19 +220,85 @@ public Object readObject() throws SQLException {
 
   @Override
   public <T> T readObject(Class<T> type) throws SQLException {
+    return readObject(type, TimeZone.getDefault());
+  }
+
+  @Override
+  public <T> T readObject(Class<T> type, TimeZone tz) throws SQLException {
```

Comment:
This is something new, are we sure we want to expose such method? In `java.sql.ResultSet` there is no such addition in `getObject`...

Reply:
It is something ekstra added in interface SFSqlInput. It gives flexibility in reading timestamps by readObject(). In my opinion it should be added to getObject as well

Reply:
Deleted. We can add this in future